tensorflow/core/platform/env.cc

```diff
@@ -361,6 +374,32 @@ Status WriteStringToFile(Env* env, const string& fname,
   return s;
 }
 
+Status CopyFile(FileSystem* src_fs, const string& src, FileSystem* target_fs,
+                const string& target) {
+  uint64 size;
+  TF_RETURN_IF_ERROR(src_fs->GetFileSize(src, &size));
+
+  std::unique_ptr<RandomAccessFile> src_file;
+  TF_RETURN_IF_ERROR(src_fs->NewRandomAccessFile(src, &src_file));
+
+  std::unique_ptr<WritableFile> target_file;
+  TF_RETURN_IF_ERROR(target_fs->NewWritableFile(target, &target_file));
+
+  uint64 offset = 0;
+  while (offset < size) {
+    char scratch[kCopyFileBufferSize];
+    StringPiece result;
+    size_t bytes_to_read =
+        offset + sizeof(scratch) < size ? sizeof(scratch) : size - offset;
```

Comment:
This is fine, but can we have an if statement instead? I always get scared when I have to search Google to make sure unsafe code is correct. Same goes for that other ternary.

Reply:
Done.